components/api-controller/pkg/controller/v1alpha2/controller.go

```diff
@@ -256,6 +265,18 @@ func (c *Controller) validateAPI(newAPI *kymaApi.Api, apiStatusHelper *ApiStatus
 	return setStatus(kymaMeta.Successful)
 }
 
+func (c *Controller) validateVirtualService(newAPI *kymaApi.Api) error {
+	for _, svc := range c.blacklistedServices {
+		separator := "."
+		props := strings.Split(svc, separator)
+
+		if newAPI.Spec.Service.Name == props[0] && newAPI.GetNamespace() == props[1] {
+			return fmt.Errorf("creating VirtualService for %s.%s is forbidden", newAPI.Spec.Service.Name, newAPI.GetNamespace())
```

Comment:
Maybe a better error would be `Service X has been blacklisted by BLACKLISTED_SERVICES env`